hack/make-rules/test-cmd.sh

```diff
@@ -116,28 +116,6 @@ function create_node() {
 __EOF__
 }
 
-# Run it if:
-# 1) $WHAT is empty
-# 2) $WHAT is not empty and kubeadm is part of $WHAT
-WHAT=${WHAT:-}
-if [[ ${WHAT} == "" || ${WHAT} =~ .*kubeadm.* ]] ; then
-  kube::log::status "Running kubeadm tests"  
-
-  # build kubeadm
-  make all -C "${KUBE_ROOT}" WHAT=cmd/kubeadm
-  # unless the user sets KUBEADM_PATH, assume that "make all..." just built it
-  export KUBEADM_PATH="${KUBEADM_PATH:=$(kube::realpath "${KUBE_ROOT}")/_output/local/go/bin/kubeadm}"
-  # invoke the tests
-  make -C "${KUBE_ROOT}" test \
-    WHAT=k8s.io/kubernetes/cmd/kubeadm/test/cmd
-
-  # if we ONLY want to run kubeadm, then exit here.
-  if [[ ${WHAT} == "kubeadm" ]]; then
-    kube::log::status "TESTS PASSED"
-    exit 0
-  fi
-fi
-
```

Comment:
noting that this change needs an update in the k/kubeadm docs:
https://github.com/kubernetes/kubeadm/blob/c4674ed58e9116933e7a6d9a613c7a3d3ba4da8e/CONTRIBUTING.md#running-unit-tests

Reply:
Hmm, I wonder if I did this correctly...

Maybe it should still be run the same way, just have the code be located in the new directory.

This really is a strange test compared to the others because it requires the kubeadm executable to be built and located.

I'm not so sure now... :/

Reply:
leaving it under `make test-cmd` seems fine.

Reply:
If test-cmd is "command line integration tests", it feels like it still belongs there.

I don't know where the right place for the code is... test/integration directory seems like it has tests for things that are not command line.

test/cmd sort of seems like where it belongs, but that directory only has shell scripts.

Reply:
OK, let me push an update, keep it in test-cmd so not cause any usage changes

Reply:
having a folder called kubeadm under `./test/cmd` seems OK to me.
leaving this decision to SIG Testing and the maintainers of /test

the alternative solution as mentioned by liggitt here:
https://github.com/kubernetes/kubernetes/issues/91740#issuecomment-638993147

is to just skip the kubeadm integration tests if the env. variable KUBEADM_PATH is not set.
and leave the kubeadm tests under cmd/kubeadm/test